### ISSUE TITLE
请问： circle loss代码怎么和论文中的公式对应起来呢

### ISSUE THREAD

**Issue by guanyonglai** [JDAI-CV/fast-reid#652]
我看了circle loss的论文还挺复杂，为啥你这里寥寥几笔就复现了circle loss呢？这里的 self.s 应该就是论文中的γ吧，这里的 sim_mat 对应论文的什么呢？论文中的公式怎么和这里代码的公式对应起来呢？我把这个项目的circle loss部分的代码单独拎出来如下：
`class CircleSoftmax(nn.Module):

    def __init__(self, cfg, in_feat, num_classes):
        super().__init__()
        self.in_feat = in_feat
        self._num_classes = num_classes
        self.s = cfg.MODEL.HEADS.SCALE # 64
        self.m = cfg.MODEL.HEADS.MARGIN # 0.35
        self.weight = Parameter(torch.Tensor(num_classes, in_feat))
        nn.init.kaiming_uniform_(self.weight, a=math.sqrt(5))
    def forward(self, features, targets):
        sim_mat = F.linear(F.normalize(features), F.normalize(self.weight))
        alpha_p = torch.clamp_min(-sim_mat.detach() + 1 + self.m, min=0.)
        alpha_n = torch.clamp_min(sim_mat.detach() + self.m, min=0.)
        delta_p = 1 - self.m  # 0.65
        delta_n = self.m
        s_p = self.s * alpha_p * (sim_mat - delta_p)
        s_n = self.s * alpha_n * (sim_mat - delta_n)
        targets = F.one_hot(targets, num_classes=self._num_classes)
        pred_class_logits = targets * s_p + (1.0 - targets) * s_n
        return pred_class_logits`
